bin/crew

```diff
@@ -1250,12 +1274,23 @@ def install
   # If this fails, the install should fail before we create any
   # damage, and we should roughly be at maximal disk space usage at this
   # point anyways.
-  FileUtils.cp File.join(CREW_CONFIG_PATH, 'device.json'), "#{CREW_CONFIG_PATH}/device.json.tmp"
+  begin
+    FileUtils.cp(File.join(CREW_CONFIG_PATH, 'device.json'), "#{CREW_CONFIG_PATH}/device.json.tmp")
+  rescue StandardError
+    puts 'Error writing installed packages json file backup!'.lightred
+    abort
+  end
 
   # remove it just before the file copy
   if @pkg.in_upgrade
     puts 'Removing since upgrade or reinstall...'
     Command.remove(@pkg, @opt_verbose)
+
+    if CREW_ESSENTIAL_PACKAGES.include?(@pkg.name)
+      crewlog "Removing package #{@pkg.name} from device.json"
+      @device_json['installed_packages'].delete_if { |entry| entry['name'] == @pkg.name }
```

Comment:
Still don't quite understand why the package needs to be removed from device.json.

Reply:
Upgrade is the union of remove and install.

Remove takes out the package's line from the json file, except when one of the `CREW_ESSENTIAL_PACKAGES` is being dealt with, in which case remove returns early and the json file never gets updated.

This works around that, as otherwise, a new line gets added to the json file
This is a problem because the version detection for packages looks for the first package info returned from the json file, and since the old info is still there after an upgrade due to this bug, crew thinks that the package still needs to be upgraded after the new package is installed.

Reply:
Removing the package should remove the package from the skin file, though.

Reply:
Ah, I see-- the removal is getting skipped, so the package isn't getting removed.